lib/rails_extensions/action_dispatch/request.rb

```diff
@@ -0,0 +1,24 @@
+module ActionDispatch
+  class Request
+    KNOWN_OPERATING_SYSTEMS = ["Windows", "Macintosh", "Linux", "Android", "iPhone"].freeze
+    KNOWN_BROWSERS = ["Chrome", "Safari", "Firefox", "Edge", "Opera"].freeze
+
+    def operating_system
+      get_item_in_str(user_agent, KNOWN_OPERATING_SYSTEMS) || "unknown operating system"
+    end
+
+    def browser
+      get_item_in_str(user_agent, KNOWN_BROWSERS) || "unknown browser"
+    end
+
+    private
+
+    def get_item_in_str(str, items)
+      items.each do |item|
+        if str.include?(item)
+          return item
+        end
+      end
+    end
```

Comment:
Ruby basically as this method build in already as find or it's alias `detect` which is a little more idiomatic for 

```suggestion
    def operating_system
      KNOWN_OPERATING_SYSTEMS.detect { |os| user_agent.include?(os) } || "unknown operating system"
    end

    def browser
      KNOWN_BROWSERS.detect { |browser| user_agent.include?(browser) } || "unknown browser"
    end
```